src/v_draw.cpp

```diff
@@ -958,6 +958,20 @@ void DCanvas::DrawLine(int x0, int y0, int x1, int y1, int palColor, uint32_t re
 #endif
 }
 
+DEFINE_ACTION_FUNCTION(_Screen, DrawLine)
+{
+	PARAM_PROLOGUE;
+	PARAM_INT(x0);
+	PARAM_INT(y0);
+	PARAM_INT(x1);
+	PARAM_INT(y1);
+	PARAM_INT(color);
```

Comment:
I would prefer to take a single color argument. You can fill palcol by doing an RGB256k table lookup.

Reply:
Gutawer - go ahead and update this and use the automap line clipping function - as long as it's not marked as private, be it by definition or by comments.

Hopefully we can get this reviewed again soon.